src/hooks/current-user.hooks.ts

```diff
@@ -0,0 +1,29 @@
+import { CurrentUserContext } from "@/contexts/current-user.context";
+import { useContext, useEffect } from "react";
+
+export function useGetCurrentUser(){
+    const {currentUser, setCurrentUser} = useContext(CurrentUserContext);
+    
+    useEffect(() => {
+        if (currentUser == null){
```

Comment:
This check would fail if the user logout and login with a different account no? I guess we just need to make sure to export the `setCurrentUser` so that we can clean up our context when we logout then.